theHarvester/__main__.py

```diff
@@ -69,6 +69,56 @@ def start():
     virtual = args.virtual_host
     word = args.domain  # type: str
 
+    def store(search_engine, source, store_host=False, store_emails=False, store_ip=False, store_people=False,
+              store_data=False, store_links=False, store_results=False):
+        search_engine.process()
+        db_stash = stash.stash_manager()
+
+        if store_host:
+            host_names = filter(search_engine.get_hostnames())
+            all_hosts.extend(host_names)
+            db_stash.store_all(word, all_hosts, 'host', source)
+        if store_emails:
+            email_list = filter(search_engine.get_emails())
+            db_stash.store_all(word, email_list, 'email', source)
+        if store_ip:
+            ips_list = search_engine.get_ips()
+            all_ip.extend(ips_list)
+            db_stash.store_all(word, ips, 'ip', source)
+        if store_data:
+            data = filter(crtsh_search.get_data())
+            all_hosts.extend(data)
+            db.store_all(word, all_hosts, 'host', source)
+        if store_results:
+            emails, hosts, urls = search_engine.get_results()
+            all_emails.extend(emails)
+            hosts = filter(hosts)
+            global trello_urls
+            trello_urls = filter(urls)
+            all_hosts.extend(hosts)
+            db.store_all(word, hosts, 'host', source)
+            db.store_all(word, emails, 'email', source)
+        if store_people:
+            people_list = search_engine.get_people()
+            db_stash.store_all(word, people_list, 'people', source)
+            if len(people_list) == 0:
+                print('\n[*] No users found.\n\n')
+            else:
+                print('\n[*] Users found: ' + str(len(people_list)))
+                print('---------------------')
+                for usr in sorted(list(set(people_list))):
+                    print(usr)
+        if store_links:
+            links = linkedin_links_search.get_links()
+            db.store_all(word, links, 'name', 'linkedin')
+            if len(links) == 0:
+                print('\n[*] No links found Linkedin.\n\n')
+            else:
+                print(f'\n[*] Links found: {len(links)}')
+                print('---------------------')
+                for link in sorted(list(set(links))):
+                    print(link)
```

Comment:
I am not to sure about having this in this function. I do have another module that I will be creating at some point soon that will be return links and that will be its soul purpose but feel like there is a better way to make it agnostic. @NotoriousRebel Thoughts on this?

Reply:
Totally agree that this should be agnostic. However, I didn't want to touch too much code around for refactoring and wanted to solely reduce those code repetitions. I can work on externalizing the method in next PR.

Will definitely add documentation and see if any unit tests can be added.